lib/github_advisory_sync.rb

```diff
@@ -199,110 +206,122 @@ def ghsa_id
+
+        return ref["url"]
       end
     end
 
-    def vulnerabilities
-      github_advisory_graphql_object["vulnerabilities"]["nodes"]
+    def package_name
+      vulnerabilities.first["package"]["name"]
     end
 
-    def has_ruby_vulnerabilities?
-      vulnerabilities.any? do |vuln|
-        vuln["package"]["ecosystem"] == "RUBYGEMS"
-      end
+    def rubysec_filename
+      File.join("gems", package_name, "#{primary_id}.yml")
     end
 
-    def some_rubysec_files_do_not_exist?
-      rubysec_filenames.any?{|filename| !File.exist?(filename) }
-    end
+    def update_file
+      saved_data = YAML.load_file(rubysec_filename)
 
-    def write_files
-      return [] unless some_rubysec_files_do_not_exist?
-
-      files_written = []
-      vulnerabilities.each do |vulnerability|
-        filename_to_write = File.join("gems", vulnerability["package"]["name"], "#{primary_id}.yml")
-        next if File.exist?(filename_to_write)
-
-        data = {
-          "gem" => vulnerability["package"]["name"],
-          "ghsa" => ghsa_id[5..],
-          "url" => external_reference,
-          "date" => published_day,
-          "title" => github_advisory_graphql_object["summary"],
-          "description" => github_advisory_graphql_object["description"],
-          "cvss_v3" => cvss,
-          "patched_versions" => [ "<FILL IN SEE BELOW>" ],
-          "unaffected_versions" => [ "<OPTIONAL: FILL IN SEE BELOW>" ]
-        }
-        data["cve"] = cve_id[4..20] if cve_id
-
-        dir_to_write = File.dirname(filename_to_write)
-        Dir.mkdir dir_to_write unless Dir.exist?(dir_to_write)
-        File.open(filename_to_write, "w") do |file|
-          # create an automatically generated advisory yaml file
-          file.write data.to_yaml
-
-          # The data we just wrote is incomplete,
-          # and therefore should not be committed as is
-          # We can not directly translate from GitHub to rubysec advisory format
-          #
-          # The patched_versions field is not exactly available.
-          # - GitHub has a first_patched_version field,
-          #   but rubysec advisory needs a ruby version spec
-          #
-          # The unnaffected_versions field is similarly not directly available
-          # This optional field must be inferred from the vulnerableVersionRange
-          #
-          # To help write those fields, we put all the github data below.
-          #
-          # The second block of yaml in a .yaml file is ignored (after the second "---" line)
-          # This effectively makes this data a large comment
-          # Still it should be removed before the data goes into rubysec
-          file.write "\n\n# GitHub advisory data below - **Remove this data before committing**\n"
-          file.write "# Use this data to write patched_versions (and potentially unaffected_versions) above\n"
-          file.write github_advisory_graphql_object.to_yaml
+      return if saved_data.key?("cvss_v3") || cvss.nil?
```

Comment:
if we're only checking to update cvss_v3, maybe we can rename the method accordingly, `update_file_cvss_v3` or something.

Reply:
Can we support updating other things? Would be nice to have an (optional) way to sync more than just CVSSv3, as if the advisories are updated, we might not pick up that at all.

Reply:
Updated to add any missing information from Github